0000-template.md

```diff
@@ -8,14 +8,26 @@ One paragraph explanation of the feature.
 
 # Motivation
 
-Why are we doing this? What use cases does it support? What is the expected outcome?
+Why are we doing this? What use cases does it support? What is the expected
+outcome?
 
 # Detailed design
 
-This is the bulk of the RFC. Explain the design in enough detail for somebody familiar
-with the framework to understand, and for somebody familiar with the implementation to implement.
-This should get into specifics and corner-cases, and include examples of how the feature is used.
-Any new terminology should be defined here.
+This is the bulk of the RFC. Explain the design in enough detail for somebody
+familiar with the framework to understand, and for somebody familiar with the
+implementation to implement. This should get into specifics and corner-cases,
+and include examples of how the feature is used. Any new terminology should be
+defined here.
+
+# Programming model
+
+How does this proposal impact the existing Ember programming model? What
+names and terminology work best for these concepts and why?
```

Comment:
"Programming model" is kind of vague and is a term I assume many contributors may not be familiar with. For me, the heuristic for whether something impacts the existing programming model is, "Would the current guides need to be updated? How significantly?" We may wish to include a question to that effect here, since I think it's easier for people to answer than "does it change the programming model?"